File: ios/perspective/Models/NewsArticle.swift
import Foundation

public struct NewsArticle: Codable, Identifiable {
    public let id: Int
    public let title: String
    public let summary: String
    public let content: String
    public let source: String
    public let sourceUrl: String?
    public let author: String?
    public let category: String
    public let imageUrl: String?
    public let publishedAt: Date
    public let credibilityScore: Double
    public let biasIndicators: [String]
    public let readingTimeMinutes: Int
    
    public enum CodingKeys: String, CodingKey {
        case id, title, summary, content, source
        case sourceUrl, author, category, imageUrl
        case publishedAt, credibilityScore, biasIndicators
        case readingTimeMinutes
    }
}

File: backend/src/models/NewsArticle.ts
export interface NewsArticle {
  id: number;
  title: string;
  content: string;
  source: string;
  author?: string;
  url: string;
  image_url?: string;
  category?: string;
  bias_rating?: number; // -3.0 to +3.0
  bias_source?: string;
  tags?: string[];
  published_at: Date;
  is_active: boolean;
  created_at: Date;
  updated_at: Date;
}

export interface ArticleWithBiasInfo extends NewsArticle {
  bias_label: string; // 'Left', 'Lean Left', 'Center', 'Lean Right', 'Right'
  reliability_score?: number;
} 